src/ngx_http_lua_ssl.c

```diff
@@ -34,4 +38,169 @@ ngx_http_lua_ssl_init(ngx_log_t *log)
 }
 
 
+static size_t
+ngx_http_lua_ssl_get_error(u_long e, u_char *ssl_err,
+    size_t ssl_err_len, const char *default_err, size_t default_err_len)
+{
+    size_t             len;
+
+    if (e == 0) {
+        len = ngx_min(ssl_err_len, default_err_len);
+        ngx_memcpy(ssl_err, default_err, len);
+
+        return len;
+    }
+
+    ERR_error_string_n(e, (char *) ssl_err, ssl_err_len);
+
+    return ngx_strlen(ssl_err);
+}
+
+
+#ifndef NGX_LUA_NO_FFI_API
+
+
+void *
+ngx_http_lua_ffi_ssl_ctx_init(ngx_uint_t protocols, char **err)
+{
+    ngx_ssl_t           ssl;
+
+    ssl.log = ngx_cycle->log;
+    if (ngx_ssl_create(&ssl, protocols, NULL) != NGX_OK) {
+        *err = "failed to create ssl ctx";
```

Comment:
Also, I think it would be better to return OpenSSL's own error messages if there's any. Otherwise fixed error messages made up in the caller does not give much helpful info.